src/cert_utils.erl

```diff
@@ -0,0 +1,87 @@
+-module(cert_utils).
+-author('denys.gonchar@erlang-solutions.com').
+
+-export([ get_cert_domains/1,
+          get_common_name/1,
+          get_xmpp_addresses/1,
+          get_dns_addresses/1
+        ]).
+
+-include_lib("public_key/include/public_key.hrl").
+-include("XmppAddr.hrl").
+-include("jlib.hrl").
+
+-type certificate() :: #'Certificate'{}.
+
+-spec get_common_name(certificate()) -> string() | error.
+get_common_name(Cert) ->
+  try
+    {rdnSequence, RDNSequence} = Cert#'Certificate'.tbsCertificate#'TBSCertificate'.subject,
+    [{ok, {_, CN}}] = ['OTP-PUB-KEY':decode('X520CommonName', V) ||
+      AtributesList <- RDNSequence,
+      #'AttributeTypeAndValue'{type = ?'id-at-commonName', value = V} <- AtributesList],
+    CN
+  catch
+    _:_ -> error
+  end.
+
+
+-spec get_xmpp_addresses(certificate()) -> [any()].
+get_xmpp_addresses(Cert) ->
+  try
+    Extensions = Cert#'Certificate'.tbsCertificate#'TBSCertificate'.extensions,
+    [BinVal] = [convert_to_bin(V) ||
+      #'Extension'{extnID = ?'id-ce-subjectAltName', extnValue = V} <- Extensions],
+    {ok, SANs} = 'OTP-PUB-KEY':decode('SubjectAltName', BinVal),
+    [begin
+       {ok, XmppAddr} = 'XmppAddr':decode('XmppAddr', V),
+       XmppAddr
+     end || {otherName, #'AnotherName'{'type-id' = ?'id-on-xmppAddr', value = V}} <- SANs]
+  catch
+    _:_ -> []
+  end.
+
+
+-spec get_dns_addresses(certificate()) -> [any()].
+get_dns_addresses(Cert) ->
+  try
+    Extensions = Cert#'Certificate'.tbsCertificate#'TBSCertificate'.extensions,
+    [BinVal] = [convert_to_bin(V) ||
+      #'Extension'{extnID = ?'id-ce-subjectAltName', extnValue = V} <- Extensions],
+    {ok, SANs} = 'OTP-PUB-KEY':decode('SubjectAltName', BinVal),
+    [DNS || {dNSName, DNS} <- SANs]
+  catch
+    _:_ -> []
```

Comment:
ok, catching everything is usually bad practice.
I should be used with care and should report errors and stacktrace.
Otherwise we would ignore some invisible errors.

Reply:
i'm not gonna log here anything, it's normal when certificate is missing some of the fields, generally  it's not an error. if you want to log something - do it in the proper place, it's a duty of the caller

Reply:
`try ... catch _:_ ` construction should not used in code, because it hides all the errors.
Both expected and not.

Reply:
If there was a typo in `SubjectAltName`, for example `SubjectAltnaen` we would never find this out.

I am fine with 

```
case decode(....) of
    {ok, SANs} ->
....
{error, _} ->
[]
end
```

On microbenchmarking side: going inside catch construction is slower than just case.

Reply:
Michael, this module will be covered with the tests. that will be done in the next PR because we need to change our SSL certs generation suit, and that is big chunk of work. generally i did manual tests for this code so it's ok to archive it. this code provides only ASN1 parsing, it is ok to to use try/catch here. i'm not gonna change it. period. this comment is related to all the parsing functions.

Reply:
It's not about doing a typo in a code or tests but lack of any meaningful error return here will make it more difficult to debug problems with real-world certificates. I'd even prefer a lack of `try..catch` rather than hiding errors.